### PR TITLE
added powered-by-header

### PR DESCRIPTION
**Pull request by luceos**
Adds a `X-Powered-By` header. It can be disabled using:

`Flarum\Http\Middleware\FlarumPromotionHeader::disable()` in extend.php.

### CODE REVIEW

**Review by KyrneDev — APPROVED** [review]
Something I did know we needed but I love it! Everything works locally from testing.

File: src/Http/Middleware/FlarumPromotionHeader.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+namespace Flarum\Http\Middleware;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface as Middleware;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class FlarumPromotionHeader implements Middleware
+{
+    public static $enabled = true;
```
Comment (askvortsov1) [inline src/Http/Middleware/FlarumPromotionHeader.php:19]: I would prefer for this to inject `Config`, and read from that instead.

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
> It can be disabled using:
> 
> `Flarum\Http\Middleware\FlarumPromotionHeader::disable()` in extend.php.

The `Compat` extender is going away with stable, so no more callbacks in extend.php. I suppose it could also be done outside a callback, but I would prefer for `extend.php` to remain just extenders. Instead, let's use the config.php.

File: src/Http/Middleware/FlarumPromotionHeader.php (hunk 1 above)
Comment (tankerkiller125) [reply src/Http/Middleware/FlarumPromotionHeader.php:19]: I agree, a config option would be much better in my opinion as well.

Comment (askvortsov1) [reply src/Http/Middleware/FlarumPromotionHeader.php:19]: I don't think an extension should be needed to toggle this, hence my preference for config.php

Comment (luceos) [reply src/Http/Middleware/FlarumPromotionHeader.php:19]: I like the idea of using config.php for this. Will refactor.

**Comment by luceos** [comment]
I've refactored based on the review. You can now disable the promotion header in the `config.php` like so:

```php
<?php

return [
    'url' => 'http://toby.local',
    'debug' => true,
    'promote' => false,
    // ..
];

File: src/Http/Middleware/FlarumPromotionHeader.php [hunk 2]
```diff
@@ -9,25 +9,26 @@
...
 class FlarumPromotionHeader implements Middleware
 {
-    public static $enabled = true;
+    protected $enabled = true;
 
-    public static function disable()
+    public function __construct(Config $config)
     {
-        static::$enabled = false;
+        $this->enabled = $config['promote'] ?? true;
```
Comment (askvortsov1) [inline src/Http/Middleware/FlarumPromotionHeader.php:24]:
```suggestion
        $this->enabled = $config['flarumPromotionalHeader'] ?? true;
```

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

**Comment by SychO9** [comment]
can we also add the key to StoreConfig.php so that it's there by default ?

**Comment by askvortsov1** [comment]
> can we also add the key to StoreConfig.php so that it's there by default ?

Good point. It'll also be in the release announcement and config.php docs

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
I think "promote" is too broad, "promoteFlarumHeader" or "promoteFlarum" might be better.

File: src/Http/Middleware/FlarumPromotionHeader.php (hunk 2 above)
Comment (tankerkiller125) [reply src/Http/Middleware/FlarumPromotionHeader.php:24]: I feel like that suggested name is a little long? I'm perfectly fine with simple "promote" if we did want to explain it more maybe "promotionHeader"

Comment (askvortsov1) [reply src/Http/Middleware/FlarumPromotionHeader.php:24]: I like it!

**Comment by KyrneDev** [comment]
> I think "promote" is too broad, "promoteFlarumHeader" or "promoteFlarum" might be better.

What about "promoHeader"

**Review by askvortsov1 — APPROVED** [review]

**Review by tankerkiller125 — APPROVED** [review]

**Pull request merged**